### PR TITLE
Pass engagementButtonLabel to showHeartbeat properly.

### PR DESCRIPTION
**Pull request by Osmose**
Also includes some refactoring to make testing the Normandy driver possible / easier.

@mythmon r?

### CODE REVIEW

File: normandy/control/static/control/js/components/RecipePreview.jsx [hunk 1]
```diff
@@ -1,7 +1,8 @@
-import React from 'react'
-import classNames from 'classnames'
-import composeRecipeContainer from './RecipeContainer.jsx'
-import {runRecipe} from '../../../../../selfrepair/static/js/self_repair_runner.js';
+import React from 'react';
+import classNames from 'classnames';
+import composeRecipeContainer from './RecipeContainer.jsx';
+import { runRecipe } from '../../../../../selfrepair/static/js/self_repair_runner.js';
+import NormandyDriver from '../../../../../selfrepair/static/js/normandy_driver.js';
```
Comment (mythmon) [inline normandy/control/static/control/js/components/RecipePreview.jsx]: 😞 We should find a way to avoid all these dots in the path. They aren't good.

**Comment by mythmon** [comment]
"some refactoring" is a bit of an understatement. Would it be possible to split the refactoring changes into a separate commit?

**Comment by Osmose** [comment]
> "some refactoring" is a bit of an understatement. Would it be possible to split the refactoring changes into a separate commit?

I could. If you view the diff with whitespace changes disabled it's actually much more manageable (I'm a monster and switched the file to 2-spaces during this commit to match the rest of the project). But if that's still too much I totally can.

File: normandy/control/static/control/js/components/RecipePreview.jsx (hunk 1 above)
Comment (Osmose) [inline normandy/control/static/control/js/components/RecipePreview.jsx]: If I could make all imports relative to the static folders I would. In fact, I think I'll look into that at some point.

Comment (brittanystoroz) [inline normandy/control/static/control/js/components/RecipePreview.jsx]: http://moduscreate.com/es6-es2015-import-no-relative-path-webpack/

Comment (mythmon) [inline normandy/control/static/control/js/components/RecipePreview.jsx]: Nice. That looks like a great idea. We'll just have to make sure none of the places we want to import from collide with any package names.

**Comment by mythmon** [comment]
Even with `?w=1`, it is hard to tease out what was a change just to refactor and what was a change in behavior. Maybe I'm just being lazy though. Plus, you can't comment in `?w=1` mode

File: normandy/selfrepair/tests/test_normandy_driver.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import NormandyDriver from '../static/js/normandy_driver.js';
+
+describe('Normandy Driver', () => {
+  describe('showHeartbeat', () => {
+    it('should pass all the required arguments to the UITour helper', async function() {
```
Comment (mythmon) [inline normandy/selfrepair/tests/test_normandy_driver.js]: Lint: prefer arrow functions. Eslint has some issues with `async () => {}`, but it is valid. If this is problematic for you, leave it and I'll fix it in my linter branch (which can deal with `async () => {}`).

File: normandy/selfrepair/static/js/self_repair.js [hunk 3]
```diff
@@ -1,16 +1,20 @@
-import Normandy from './normandy_driver.js';
+import Mozilla from './uitour.js';
+import NormandyDriver from './normandy_driver.js';
 import {fetchRecipes, filterContext, doesRecipeMatch, runRecipe} from './self_repair_runner.js';
 
+let driver = new NormandyDriver(Mozilla.UITour);
```
Comment (mythmon) [inline normandy/selfrepair/static/js/self_repair.js]: This is just the default value, right?

**Comment by mythmon** [comment]
LGTM, besides a couple nits and much easier to review in those chunks. Thanks for breaking it up.

r+

File: normandy/selfrepair/static/js/self_repair.js (hunk 3 above)
Comment (Osmose) [inline normandy/selfrepair/static/js/self_repair.js]: Yeah but I wanted to be explicit about it. I dunno.

**Comment by Osmose** [comment]
@mythmon Had to rebase this after the linting changes and it was sufficiently complex that another quick glance over this would be appreciated. :D

**Comment by mythmon** [comment]
Nice!

**Pull request merged**